app/models/observation.rb

```diff
@@ -193,6 +193,50 @@ class Observation < AbstractModel
   # Automatically (but silently) log destruction.
   self.autolog_events = [:destroyed]
 
+  include CreatedUpdatedScopes
+
+  scope :include_all_name_proposals, lambda { |name|
+    joins(:namings).where(namings: { name: name })
+  }
+  scope :of_look_alikes, lambda { |name|
+    joins(:namings).where(namings: { name: name }).
+      where(Observation[:name] != name)
+  }
+  scope :include_subtaxa, lambda { |text_name|
+    names = {}
+    starting_names = Name.text_name_like(text_name).with_correct_spelling
+    starting_names.each do |name|
+      names = starting_names.or(Name.related_taxa(name)).distinct
+    end
+    where(name: names)
+  }
+  scope :of_synonyms, ->(name) { where(name: name.synonyms) }
+  scope :of_name, ->(name) { where(name: name) }
+  scope :of_name_like,
+        ->(name) { where(name: Name.text_name_like(name)) }
+  scope :with_name, -> { where.not(name: Name.unknown) }
+  scope :without_name, -> { where(name: Name.unknown) }
+  scope :by_user, ->(user) { where(user: user) }
+  scope :at_location, ->(location) { where(location: location) }
+  scope :in_region,
+        ->(where) { where(Observation[:where].matches("%#{where}%")) }
+  scope :is_collection_location, -> { where(is_collection_location: true) }
+  scope :not_collection_location, -> { where(is_collection_location: false) }
+  scope :for_project, lambda { |project|
+    joins(:project_observations).
+      where(ProjectObservation[:project_id] = project)
+  }
+  scope :with_image, -> { where.not(thumb_image: nil) }
+  scope :without_image, -> { where(thumb_image: nil) }
+  scope :notes_like,
```

Comment:
(I called this "notes_has" or something like that in Query, didn't I?  I'm sorry, I just balk at the use of "like" specifically because "like" means something very specific in mysql, and that's only sort of what this scope is doing.  Yes, it's using "LIKE" but it's using a customized form of LIKE that doesn't allow you to anchor the pattern to the beginning or end of the string, so I think it's a subtle but important misrepresentation.

Reply:
Ok if we go with `notes_include` just to keep the scope pattern easier to remember?